### DISCUSSION TITLE
Conditionally change a field

### DISCUSSION THREAD

**Discussion by mbeynon** [datanymizer/datanymizer#141] [Q&A] [answered]
I need to transform all rows of a table.  Some fields are always transformed.  One field is a **message** field, which can be _NULL_ or a string such as _"Hello world"_.

Is it possible to conditionally anonymize this field in the table, so it retains the proportion of those that were _NULL_ vs those with a value?

Specifically, I'd like to leave all the _NULL_ values as is, and only anonymize with a new value if it previously was a string (not _NULL_).  The replacement would either be fixed in value such as _"This is a test"_ or randomized by generating a few words.

The important part is making that conditional check if the original value was _NULL_.  Can this be done with the current version of datanymizer?

Thank you.

**Comment by mbeynon** [comment, marked as answer]
Looks like you can do it using the Tera conditionals.  Was a little unclear how to test for NULL, but this seems to work:

        template:
          # prev.message can be NULL
          format: '{% if prev.message != "\N" %}NEW VALUE{% endif %}'

**Reply by evgeniy-r** [reply]
> Is it possible to conditionally anonymize this field in the table, so it retains the proportion of those that were NULL vs those with a value?

Actually, we are planning to implement something like this.
